File: searched-common/src/main/java/com/songpo/searched/util/MD5Util.java
package com.songpo.searched.util;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class MD5Util {

    private static final String[] HEX_DIGITS = {"0", "1", "2", "3", "4", "5",
            "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"};

    public static String byteArrayToHexString(byte b[]) {
        StringBuffer resultSb = new StringBuffer();
        for (byte aB : b) {
            resultSb.append(byteToHexString(aB));
        }

        return resultSb.toString();
    }

    // 返回形式为数字跟字符串
    public static String byteToHexString(byte b) {
        int n = b;
        if (n < 0) {
            n += 256;
        }
        int d1 = n / 16;
        int d2 = n % 16;
        return HEX_DIGITS[d1] + HEX_DIGITS[d2];
    }

    public static String md5encode(String origin, String charsetname) {
        String resultString = null;
        try {
            resultString = new String(origin);
            MessageDigest md = MessageDigest.getInstance("MD5");
            if (charsetname == null || "".equals(charsetname)) {
                resultString = byteArrayToHexString(md.digest(resultString
                        .getBytes()));
            } else {
                resultString = byteArrayToHexString(md.digest(resultString
                        .getBytes(charsetname)));
            }
        } catch (Exception exception) {
        }
        return resultString;
    }
    public static String GetMD5Code(String string,String key) {
        string=string+"&key="+key;
        String resultString = null;
        try {
            resultString = new String(string);
            MessageDigest md = MessageDigest.getInstance("MD5");
            // md.digest() 该函数返回值为存放哈希值结果的byte数组
            resultString = byteToString(md.digest(string.getBytes()));
        } catch (NoSuchAlgorithmException ex) {
            ex.printStackTrace();
        }
        return resultString;
    }
    // 转换字节数组为16进制字串
    private static String byteToString(byte[] bByte) {
        StringBuffer sBuffer = new StringBuffer();
        for (int i = 0; i < bByte.length; i++) {
            sBuffer.append(byteToHexString(bByte[i]));
        }
        return sBuffer.toString();
    }
}

File: searched-common/src/main/kotlin/com/songpo/searched/wxpay/util/MD5Util.kt
package com.songpo.searched.wxpay.util

import java.security.MessageDigest

object MD5Util {

    private val hexDigits = arrayOf("0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f")
    private fun byteArrayToHexString(b: ByteArray): String {
        val resultSb = StringBuffer()
        for (aB in b) {
            resultSb.append(byteToHexString(aB))
        }

        return resultSb.toString()
    }

    private fun byteToHexString(b: Byte): String {
        var n = b.toInt()
        if (n < 0) {
            n += 256
        }
        val d1 = n / 16
        val d2 = n % 16
        return hexDigits[d1] + hexDigits[d2]
    }

    fun MD5Encode(origin: String, charsetname: String?): String? {
        var resultString: String? = null
        try {
            resultString = origin
            val md = MessageDigest.getInstance("MD5")
            if (charsetname == null || "" == charsetname) {
                resultString = byteArrayToHexString(md.digest(resultString
                        .toByteArray()))
            } else {
                resultString = byteArrayToHexString(md.digest(resultString
                        .toByteArray(charset(charsetname))))
            }
        } catch (exception: Exception) {
        }

        return resultString
    }

}

